vignettes/secuTrialR-package-vignette.Rmd

```diff
@@ -1,172 +1,570 @@
+```
 
-### Analysing data completeness
+Please note that earlier R versions may return different results because
+there is a different `rng_config`.
+For this reason we have added the `rng_config` to the output.
 
-For various reasons eCRFs are frequently only partially filled. This can be commonly
-observed in cohort studies of registries. Thus, before performing more thorough analyses
-it is sometimes informative to get an overview of data completeness.
+## Retrieve score variables
 
-To asses data completeness of a secuTrial export you first need to generate the
-validation overview in secuTrial and export it as an *.xlsx file. Remember to select the
-"Column" and "Completion status" columns before you export the validation overview.
-The validation overview for this vignette is also part of this package.
+Generally, it is advisable to recalculate score variables before data analysis.
+A rule of thumb: The more complex a score is and the more data from different 
+forms is necessary for its calculation the more likely its value should be recalculated.
+The below function will allow you to detect which variables this concerns.
+\vspace{5pt}
 
 ```{r}
-val_ovv_location <- system.file("extdata", "sT_exports", "BMD",
-                                "bmd_validation_overview.xlsx", 
-                                package = "secuTrialR")
+return_scores(ctu05_data)
+```
+
+\newpage
+
+## Finding changes/differences in exports
+
+In ongoing studies it is possible that changes to the secuTrial data entry
+interface (i.e. the electronic case report forms) are made. Sometimes these
+changes may call for adjustments in analysis code. It is considered good practice
+to run  `diff_secuTrial()` on the last export and the current export of a project
+to at least make yourself aware of potential changes in the setup. If there are
+differences, the results of this function should be interpreted as a first indicator
+since they may not cover all alterations. Information is returned on new forms and
+variables. A detailed list of changes can be produced in the secuTrial
+FormBuilder with "Compare project setup".
+\vspace{5pt}
+
+``` {r}
+# load second export from the same project
+export_location <- system.file("extdata", "sT_exports", "longnames",
+                               "s_export_CSV-xls_CTU05_long_ref_miss_en_utf8.zip",
+                               package = "secuTrialR")
+# read all export data
+ctu05_data_new <- read_secuTrial_raw(data_dir = export_location)
+
+# show diff
+diff_secuTrial(ctu05_data, ctu05_data_new)
+
 ```
 
-In order to asses variable completeness we first require a loaded secuTrial
-validation overview and a loaded secuTrial export. Both the validation
-overview and the data export should be from the same time (not more than minutes
-apart if possible) to ensure data integrity.
+This `list` contains only empty entries. Thus, you can conclude that nothing has changed.
```

Comment:
At some point in the future we might want to add an example where something has changed.

Reply:
I agree.